src/server/raft_client.rs

```diff
@@ -822,21 +830,41 @@ async fn start<S, R, E>(
                 continue;
             }
         };
-        let client = back_end.connect(&addr);
+        let channel = back_end.connect(&addr);
+        if !channel.wait_for_connected(Duration::from_secs(1)).await {
```

Comment:
This is too short. I have seen just the name resolve takes more than 3 seconds in production environment.